### PR TITLE
Add prefer object spread instead of object assign.(closes #30)

### LINKED ISSUE #30: Prefer object spread instead of Object.assign
Now that node 8.9 has object spread, we can activate https://www.npmjs.com/package/eslint-plugin-prefer-object-spread.

### PR DESCRIPTION
**Pull request by AfonsoVReis**
Solution proposal to close #30  - "Prefer object spread instead of Object.assign() .

Added the plugin with the desired rule for the MOXY eslint configuration.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,8 +1,11 @@
 {
     "root": true,
     "extends": [
-        "./es5.js",
```
Comment (satazor) [inline .eslintrc.json]: Why was this file changed? This file is the configuration of eslint to use in itself.

File: addons/object-spread.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
-'use strict';
```
Comment (satazor) [inline addons/object-spread.js]: You will need to revert this once you revert the `.eslintrc` file.

File: rules/object-spread.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// eslint-plugin-prefer-object-spread  - https://github.com/bryanrsmith/eslint-plugin-prefer-object-spread
```
Comment (satazor) [inline rules/object-spread.js]: There's no need to create have this file.

Could you move the activation of the rule into `addons/object-spread` instead? Just create a `rules` property and put the rule inside it.

File: test/fixtures/rules/object-spread/object-spread.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// `prefer-object-spread/prefer-object-spread` - Require use of object spread instead of object.assign
```
Comment (satazor) [inline test/fixtures/rules/object-spread/object-spread.js]: Following my suggestion to remove `rules/object-spread`, please create a `contrived.bad.js` inside https://github.com/moxystudio/eslint-config/tree/master/test/fixtures/samples/es7-object-spread instead of having this file.

**Review by satazor — CHANGES_REQUESTED** [review]

File: test/.eslintrc.json [hunk 5]
```diff
@@ -3,6 +3,7 @@
     "extends": [
         "../es6.js",
         "../addons/node.js",
-        "../addons/jest.js"
+        "../addons/jest.js",
+        "../addons/object-spread.js"
```
Comment (satazor) [inline test/.eslintrc.json]: I don't think we need this change either since it's already activated in the sample, see: https://github.com/moxystudio/eslint-config/blob/master/test/fixtures/samples/es7-object-spread/.eslintrc.json#L6

File: test/__snapshots__/index.spec.js.snap [hunk 6]
```diff
@@ -840,14 +840,14 @@ Array [
     "severity": 2,
   },
   Object {
-    "column": 5,
```
Comment (satazor) [inline test/__snapshots__/index.spec.js.snap]: Do you know why was this snapshot changed? Seems unrelated.

File: test/fixtures/samples/es7-object-spread/contrived.bad.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+// Made up sample.. not genuine
```
Comment (satazor) [inline test/fixtures/samples/es7-object-spread/contrived.bad.js:1]: Could you add to the `contrived.good.js` file, an example where Object.assign is still allowed? e.g.:

```js
const a = { b: 1 };

Object.assign(a, { c: 2 });
```

**Review by satazor — CHANGES_REQUESTED** [review]

File: test/__snapshots__/index.spec.js.snap (hunk 6 above)
Comment (AfonsoVReis) [reply test/__snapshots__/index.spec.js.snap]: I was having trouble rebuilding the snapshot file within my windows client, so i had to change to ubuntu.
I will definitly take a look at it and report.

Comment (satazor) [reply test/__snapshots__/index.spec.js.snap]: Ok, I guess the build is failing because of this issue.

**Comment by satazor** [comment]
After everything is done, please squash your commits into a single one and follow https://conventionalcommits.org/ guidelines.

Additionally, this is a breaking change because code that doesn't use spread will not pass linting. Therefore, be sure to include `BREAKING CHANGE: <reason>` in the body of the commit.

File: test/__snapshots__/index.spec.js.snap (hunk 6 above)
Comment (AfonsoVReis) [reply test/__snapshots__/index.spec.js.snap]: A fresh install of node fixed the issue, now building as expected.

File: test/__snapshots__/index.spec.js.snap [hunk 8]
```diff
@@ -1701,7 +1701,39 @@ Array [
...
+  Object {
+    "column": 13,
+    "line": 7,
+    "rule": "prefer-object-spread/prefer-object-spread",
+    "severity": 2,
+  },
+]
+`;
+
+exports[`should pass on fixtures/samples/es7-object-spread/contrived.good.js 1`] = `
```
Comment (satazor) [inline test/__snapshots__/index.spec.js.snap]: This snapshot should have 0 errors. Simple use the variables in a `console.log` statement so that the errors go away. It also seems that there are some undeclared variables. Make sure that the `contrived.good` sample have 0 errors.

File: test/__snapshots__/index.spec.js.snap [hunk 9]
```diff
@@ -1701,7 +1701,39 @@ Array [
 
 exports[`should pass on fixtures/samples/es7/contrived.good.js 1`] = `Array []`;
 
-exports[`should pass on fixtures/samples/es7-object-spread/contrived.good.js 1`] = `Array []`;
+exports[`should pass on fixtures/samples/es7-object-spread/contrived.bad.js 1`] = `
+Array [
+  Object {
+    "column": 7,
+    "line": 7,
+    "rule": "no-unused-vars",
```
Comment (satazor) [inline test/__snapshots__/index.spec.js.snap]: This snapshot should not contain this error. Simply add a `console.log` statement to make use of that variable.

**Review by satazor — CHANGES_REQUESTED** [review]

**Review by satazor — APPROVED** [review]

**Comment by satazor** [comment]
All good. Could you please squash all the commits into a single one?

**Comment by satazor** [comment]
Be sure to include `Closes #30` also in the message body, besides the `BREAKING CHANGE`.

**Comment by satazor** [comment]
👍
